query-grammar/src/query_grammar.rs

```diff
@@ -300,10 +300,42 @@ fn term_group_infallible(i: &str) -> JResult<&str, UserInputAst> {
     }
 }
 
+fn exists(i: &str) -> IResult<&str, UserInputLeaf> {
```

Comment:
Can we stop the one char variable? If it is `i` for input, let's at least go for `inp`